daffodil-lib/src/main/scala/org/apache/daffodil/api/Validator.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.api
+
+import org.xml.sax.ErrorHandler
+
+trait Validator {
+  def name(): String = getClass.getTypeName.toLowerCase
+
+  def checkArgs(args: Validator.Arguments): Either[String, Unit] = Right(())
```

Comment:
If a validator needs arguments specified then it needs them at all entry points.  But I see what you are saying.

The question would be then how to separate the validation of arguments from the implementation of the validator?  Having them separate might organize better in the API but seems like it would be a burden to implement argument checking then.

Maybe a second interface that is an opt-in for argument checking... validating the validator :exploding_head:

Reply:
Moved this logic to `Validator.CompilerOps.CheckArgs` the idea being that an implementation can still provide guidance to how arguments should be formed and that guidance is consumed as part of the compilation process.